executor.go

```diff
@@ -47,17 +46,33 @@ type Executor struct {
 	Cluster *Cluster
 
 	// Client used for remote requests.
-	client InternalClient
+	client InternalQueryClient
 
 	// Maximum number of SetBit() or ClearBit() commands per request.
 	MaxWritesPerRequest int
 }
 
+type ExecutorOpt func(e *Executor) error
+
+func ExecutorOptInternalQueryClient(c InternalQueryClient) ExecutorOpt {
```

Comment:
I think that elsewhere we've used "Opt<Thing><option>" pattern rather than <Thing>Opt<option>